engine/utils/file_logger.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <trantor/utils/AsyncFileLogger.h>
+#include <trantor/utils/Utilities.h>
+#include <deque>
+
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <unistd.h>
+#endif
+
+namespace trantor {
+
+class TRANTOR_EXPORT FileLogger : public AsyncFileLogger {
+ public:
+  FileLogger();
+  ~FileLogger();
+
+  /**
+     * @brief Set the maximum number of lines to keep in the log file.
+     *
+     * @param maxLines
+     */
+  void setMaxLines(uint64_t maxLines) { max_lines_ = maxLines; }
+
+  /**
+     * @brief Set the log file name.
+     *
+     * @param fileName The full name of the log file.
+     */
+  void setFileName(const std::string& fileName) {
+    filePath_ = "./";
+    fileBaseName_ = fileName;
+    fileExtName_ = "";
+  }
+  void output_(const char* msg, const uint64_t len);
+
+ protected:
+  class CircularLogFile {
+   public:
+    CircularLogFile(const std::string& fileName, uint64_t maxLines);
+    ~CircularLogFile();
+
+    void writeLog(const char* logLine, const uint64_t len);
+    void flush();
+    uint64_t getLength() const { return totalLines_; }
+
+   private:
+    FILE* fp_{nullptr};
+    uint64_t max_lines_;
+    uint64_t totalLines_{0};
```

Comment:
I'm not sure why we need `totalLines_` variable. Can you please elaborate?

Reply:
Seem like we don't need it, I removed `totalLines_`